commons/src/main/java/gov/cms/qpp/conversion/model/validation/SubPopulationLabel.java

```diff
@@ -0,0 +1,100 @@
+package gov.cms.qpp.conversion.model.validation;
+
+import com.google.common.collect.Sets;
+
+import java.util.Arrays;
+import java.util.Set;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+/**
+ * Construct that helps classify SubPopulation types.
+ */
+public enum SubPopulationLabel {
+	DENEXCEP(SubPopulation::getDenominatorExceptionsUuid, "DENEXCEP"),
+	DENEX(SubPopulation::getDenominatorExclusionsUuid, "DENEX"),
+	NUMER(SubPopulation::getNumeratorUuid, "NUMER"),
+	DENOM(SubPopulation::getDenominatorUuid, "DENOM"),
+	IPOP(SubPopulation::getNumeratorUuid, "IPOP", "IPP");
+
+	private Set<String> aka;
+	private Function<SubPopulation, String> getter;
+
+	/**
+	 * Initialize enum constituents with functions to retrieve respective sub-population ids
+	 * along with aliases to contextualize the retrieval
+	 *
+	 * @param getter {@link SubPopulation} getter
+	 * @param aliases determines to which sub-population aliases the getter applies
+	 */
+	SubPopulationLabel(Function<SubPopulation, String> getter, String... aliases) {
+		this.getter = getter;
+		this.aka = Sets.newHashSet(aliases);
+	}
+
+	/**
+	 * Retrieve the associated aliases.
+	 *
+	 * @return an array of aliases
+	 */
+	public String[] getAliases() {
```

Comment:
why return an array?

Reply:
Convenience, the return value for this is provided as arguments to a couple of methods that accept varargs.